collectors/bzimport/convertors.py

```diff
@@ -585,6 +585,20 @@ def major_incident_state(self):
 
         return valid_pairs.get(flags_from_bz, Flaw.FlawMajorIncident.INVALID)
 
+    @cached_property
+    def major_incident_start_dt(self):
+        """
+        If the flaw is an approved major incident, the start date is assumed to be
+        the date when the hightouch flag was created.
+        """
+        if self.major_incident_state in {
+            Flaw.FlawMajorIncident.APPROVED,
+            Flaw.FlawMajorIncident.CISA_APPROVED,
+        }:
+            for flag in self.flags:
+                if flag["name"] == "hightouch":
+                    return flag["creation_date"]
```

Comment:
There is one tricky thing. In Bugzilla the regular Major Incident which when confirmed stands for `APPROVED` here is represented by `hightouch` flag there and the CISA Major Incident which when confirmed stands for `CISA_APPROVED` here is represented by `hightouch_lite` flag there. That means that for each of the two states you have to get the date from a different Bugzilla flag (the other is usually not existing). Also there may be rare but existing situation when both flags are raised - which is not really valid but we have to handle it somehow. In such case we prioritize the regular Major Incident (`hightouch`) and ignore the CISA one.

Reply:
Actually I am wrong on the invalid case. We do not prioritize anything in this scenario but simply use an `INVALID` Major Incident state. You can do whatever you want I guess :smile: